config/git/src/main/kotlin/GitConfigFileProvider.kt

```diff
@@ -104,11 +121,13 @@ class GitConfigFileProvider internal constructor(
         synchronized(this) {
             // TODO: There might be a better way to do check if the configDir already contains a Git repository.
             val revision = if (!configDir.resolve(".git").isDirectory) {
-                val initRevision = requestedRevision.takeUnless { it.isEmpty() } ?: git.getDefaultBranchName(gitUrl)
-                val vcsInfo = VcsInfo(VcsType.GIT, gitUrl, initRevision)
+                withAuthenticator(username, token) {
```

Comment:
Would it be possible to simply add the credentials to the Git URL? Then the whole authenticator implementation would not be needed.

Reply:
Just some random thoughts:
- Would that work for all URL schemes? E.g. for SSH, always the generic `git` username needs to be used instead of the real one.
- Do we need to be careful to not log / store the URL with credentials then?

Reply:
* Would it be possible to simply add the credentials to the Git URL? Then the whole authenticator implementation would not be needed.

I don't know, I have not tried it because it did not appear as a good option to me. Isn't the plain password saved in `.git/config` then?